testplan/common/utils/match.py

```diff
@@ -296,22 +538,54 @@ def get_between(self, mark1=None, mark2=None):
         :return: The content between mark1 and mark2.
         :rtype: ``str``
         """
-        if mark1 is not None and mark2 is not None:
-            if self.marks[mark1] >= self.marks[mark2]:
-                raise ValueError(
-                    'Mark "{}" must be present before mark "{}"'.format(
-                        mark1, mark2
+        try:
+            log_handle = None
+
+            if mark1 is not None and mark2 is not None:
+                if self.strategy.invalid_position_order(
+                    start_position=self.marks[mark1],
+                    end_position=self.marks[mark2],
+                ):
+                    raise ValueError(
+                        f'Mark "{mark2}" must be present before mark "{mark1}"'
                     )
-                )
 
-        with open(self.log_path, "r") as log:
-            start_pos = self.marks[mark1] if mark1 is not None else 0
-            end_pos = self.marks[mark2] if mark2 is not None else None
-            log.seek(start_pos)
-            if not end_pos:
-                return log.read()
             lines_between = []
-            while end_pos > log.tell():
-                line = log.readline()
-                lines_between.append(line)
+            start_position = (
+                self.marks[mark1]
+                if mark1 is not None
+                else self.strategy.get_start_of_file_position()
+            )
+
+            end_position = (
+                self.marks[mark2]
+                if mark2 is not None
+                else self.strategy.get_end_of_file_position()
+            )
+
+            log_handle = self.strategy.open_logfile_in_position(
+                position=start_position, mode="r"
+            )
+
+            while self.strategy.position_not_passed(
```

Comment:
@kn-ms Is there a need to remove this function? Because I have maintained it just that we will not be passing any file_handle.